### PR TITLE
Adding first pass at support for flighting.

### PR DESCRIPTION
**Pull request by EllenPorter**

### CODE REVIEW

File: node/hubController.js [hunk 1]
```diff
@@ -80,8 +80,13 @@ class HubController {
     /**
      * given a specific hub info, does the onboarding given the onboardingInfo and returns the auth info
      */
-    onboard(hubId, onboardingInfo, logger) {
+    onboard(hubId, onboardingInfo, logger, flightInfo) {
```
Comment (muralidhars) [inline node/hubController.js]: >flightInfo [](start = 43, length = 10)

validate that it is an object or null

File: node/hubController.js [hunk 2]
```diff
@@ -80,8 +80,13 @@ class HubController {
...
         logger.verbose("onboard()");
+
+        if (onboardingInfo) {
+            onboardingInfo.flights = flightInfo;
```
Comment (muralidhars) [inline node/hubController.js]: >flightInfo [](start = 37, length = 10)

is this an array? validate

File: node/hubController.js (hunk 1 above)
Comment (EllenPorter) [reply node/hubController.js]: we will want to assign flight info to null if null comes in to clear stale data.

File: node/hubController.js [hunk 3]
```diff
@@ -47,7 +47,7 @@ class HubController {
         // load info for the current hub
         var hubInfo = hubs[i];
         logger.verbose("i: " + i);
-        
+
         var LocalPackageSourceClass = require('opent2t/package/LocalPackageSource').LocalPackageSource;
```
Comment (muralidhars) [inline node/hubController.js]: nit. remove spaces

File: node/hubController.js [hunk 4]
```diff
@@ -289,6 +330,20 @@ class HubController {
...
+
+    /**
+     * Returns an array of strings
+     *  If flightInfo is a string list (expected) "a,b" = ["a", "b"]
+     *  If flightInfo is a primitive true = ["true"]
+     *  If flight info is an type array [1, "b", null, undefined, [4, 5]] = ["1", "b", "4", "5"]
+     *  If flight info is undefined = []
+     *  If flight info is null = []
+     */
+    _getFlights(flightInfo) {
```
Comment (muralidhars) [inline node/hubController.js]: >_getFlights [](start = 4, length = 11)

static _getFlights

File: node/hubController.js [hunk 5]
```diff
@@ -313,32 +368,32 @@ class HubController {
...
         }
     }
-    
+
     _setProperty(opent2t, translatorName, deviceInfo, property, deviceId, value) {
         return this._createTranslator(opent2t, translatorName, deviceInfo).then(translator => {
             return opent2t.invokeMethodAsync(translator, "", property, [deviceId, value]);
         });
     }
-
```
Comment (muralidhars) [inline node/hubController.js]: nit. remove spaces

File: node/hubController.js [hunk 6]
```diff
@@ -289,6 +329,16 @@ class HubController {
...
      * helper methods
      */
+
+    /**
+     * Returns an array of strings, expected input is string:
+     *  "flight1,flight2"
+     */
+    _getFlights(flightInfo) {
+        return (flightInfo === undefined || flightInfo === null) ? [] :
+            flightInfo.toString().split(",").filter(n => n.length > 0);
```
Comment (muralidhars) [inline node/hubController.js]: >.toString() [](start = 22, length = 11)

remove .tostring()

Comment (muralidhars) [inline node/hubController.js]: >n => n.length [](start = 52, length = 13)

unless you validate you cannot guarantee n is not null or undefined. check again here

Comment (EllenPorter) [reply node/hubController.js]: toString() converts null or undefined elements to empty strings, n cannot be null or undefined.

Comment (muralidhars) [reply node/hubController.js]: makes sense...you don't need the tostring(), split does that already

---
In reply to: [146024031](https://github.com/openT2T/cloud/pull/34#discussion_r146024031) [](ancestors = 146024031)

File: node/hubController.js [hunk 7]
```diff
@@ -310,6 +354,16 @@ class HubController {
     /**
      * helper methods
      */
+
+    /**
+     * Returns an array of flights or null
+     * expected input is string: "flight1,flight2"
+     */
+    _getFlights(flightInfo) {
+        return (flightInfo === undefined || flightInfo === null) ? null :
```
Comment (muralidhars) [inline node/hubController.js]: >flightInfo === null [](start = 44, length = 19)

typeof flightInfo === "string"

**Pull request merged**